exporter/datadogexporter/factory.go

```diff
@@ -99,23 +100,24 @@ func createMetricsExporter(
 		return nil, err
 	}
 
+	ctx, cancel := context.WithCancel(ctx)
 	var pushMetricsFn exporterhelper.PushMetrics
 
 	if cfg.OnlyMetadata {
-		pushMetricsFn = func(context.Context, pdata.Metrics) (int, error) {
-			// if only sending metadata ignore all metrics
+		pushMetricsFn = func(_ context.Context, md pdata.Metrics) (int, error) {
```

Comment:
Codecov is complaining, probably because this function (and the one for traces) never gets called in tests. We could add a test that makes sure this is called, with a mock server that receives the metadata payload.

Reply:
Done!